tests/test_exact_timing.R

```diff
@@ -444,10 +687,162 @@ climdex.pcic.test.na.masks.gsl <- function() {
+  tmin[na_indices] <- NA
+  prec[na_indices] <- NA
+  
+  # Create climdexInput object
+  ci.types.test <- climdexInput.raw(tmax = tmax, tmin = tmin, prec = prec, tmax.dates = dates, tmin.dates = dates, prec.dates = dates)
+  test.indices <- names(climdex.min.max.idx.list)
+  for(idx in test.indices){
+    fun <- paste("climdex", idx, sep = ".")
+    date.factors <- c("annual", "monthly", "seasonal")
+    for (freq in date.factors) {
+      result.e.d <- do.call(fun, list(ci.types.test, freq = freq, include.exact.dates = TRUE))
+      result.n.d <- do.call(fun, list(ci.types.test, freq = freq, include.exact.dates = FALSE))
+      checkTypes(result.e.d$val, result.n.d)
+    }
+  }
+
+  freq<- "annual"
+  result.e.d <- climdex.cdd(ci.types.test, spells.can.span.years = F, include.exact.dates = TRUE)
+  result.n.d <- climdex.cdd(ci.types.test, spells.can.span.years = F, include.exact.dates = F)
+  checkTypes(result.e.d$duration, result.n.d)
+  result.e.d <- climdex.cwd(ci.types.test, spells.can.span.years = F, include.exact.dates = TRUE)
+  result.n.d <- climdex.cwd(ci.types.test, spells.can.span.years = F, include.exact.dates = F)
+  checkTypes(result.e.d$duration, result.n.d)
+  result.e.d <- climdex.gsl(ci.types.test,"GSL", include.exact.dates = TRUE)
+  result.n.d <- climdex.gsl(ci.types.test,"GSL", include.exact.dates = F)
+  checkTypes(result.e.d$sl, result.n.d)
+}
+# 
+# climdex.pcic.test.indicies.on.random.datasets <- function(){
+#   for (i in 1:5000){
+#     cal <- 365
+#     test.dates <- seq(as.PCICt("1961-01-01", cal = cal), as.PCICt("1967-12-31", cal = cal), by = "days")
+#     test.prec.ran <- c(sample(0:2, length(test.dates), replace = TRUE))
+#     test.tmin.ran <- c(sample(-10:32, length(test.dates), replace = TRUE))
+#     test.tmax.ran <- c(sample(-10:32, length(test.dates), replace = TRUE))
+# 
+# 
+#     years <- format(test.dates, "%Y")
+#     unique.years <- unique(years)
+#     for (year in unique.years) {
+#       year.indices <- which(years == year)
+#       na.indices <- sample(year.indices, sample(1:7, 1))
+#       test.prec.ran[na.indices] <- NA
+#       test.tmin.ran[na.indices] <- NA
+#       test.tmax.ran[na.indices] <- NA
+#     }
+# 
+# 
+#     ci.ran <- climdexInput.raw(tmin =test.tmin.ran, tmax = test.tmax.ran, prec = test.prec.ran, tmin.dates=test.dates, tmax.dates = test.dates, prec.dates = test.dates)
+# 
+#     test.indices <- names(climdex.min.max.idx.list)
+#     for(idx in test.indices){
+#       fun <- paste("climdex", idx, sep = ".")
+#       date.factors <- c("annual", "monthly", "seasonal")
+#       for (freq in date.factors) {
+#         do.call(fun, list(ci.ran, freq = freq, include.exact.dates = TRUE))
+# 
+#       }
+#     }
+# 
+#     freq<- "annual"
+#     climdex.cdd(ci.ran, spells.can.span.years = F, include.exact.dates = TRUE)
+#     climdex.cwd(ci.ran, spells.can.span.years = F, include.exact.dates = TRUE)
+#     climdex.gsl(ci.ran,"GSL", include.exact.dates = TRUE)
+#   }
+# 
+# 
+# }
```

Comment:
Suggest you either add a comment to this saying why this is commented out and why/when it might be uncommented in future, OR remove the commented out code.